neuros/board/stream_processor.py

```diff
@@ -0,0 +1,136 @@
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Iterator, Optional, Callable, Any
+import numpy as np
+import threading
+import time
+from brainflow.board_shim import BoardShim, BrainFlowInputParams, BoardIds
+
+# Global lock for thread-safe board operations
+_board_lock = threading.Lock()
+
+
+@contextmanager
+def board_stream(board_id: int = BoardIds.SYNTHETIC_BOARD) -> Iterator[BoardShim]:
+    """
+    Context manager for safe board streaming operations.
+
+    Usage:
+        with board_stream() as board:
+            data = board.get_current_board_data(100)
+    """
+    board = BoardShim(board_id, BrainFlowInputParams())
+    try:
+        board.prepare_session()
+        board.start_stream()
+        yield board
+    finally:
+        with _board_lock:
+            if board.is_prepared():
+                board.stop_stream()
+                board.release_session()
+
+
+@dataclass
+class ProcessorConfig:
+    """Configuration for the stream processor"""
+    window_size_ms: float = 50.0  # Size of processing window in milliseconds
+    overlap_ms: float = 0.0  # Overlap between windows in milliseconds
+
+
+class StreamProcessor:
```

Comment:
How many of these am I allowed to create and point at the same board?

Reply:
Currently, there's no limit, and that's bad. I'll freeze it.